javascript/modules/client/src/client.ts

```diff
@@ -179,6 +188,33 @@ export class Client extends ClientCore implements IClient {
   };
 
   encoding = {
+    /**
+     * Computes the payload to be given when creating a proposal that grants a permission in a dao
+     *
+     * @param {IPermissionParams} params
+     * @return {*}  {Promise<DaoAction>}
+     * @memberof Client
+     */
+    grantAction: (params: IPermissionParams): Promise<DaoAction> =>
```

Comment:
Why a Promise?

Reply:
You need to wait for the name resolution in case we are using an ENS domain